patches/@credo-ts__openid4vc@0.5.9.patch

```diff
@@ -0,0 +1,275 @@
++        "The X509 certificate issuer must be a HTTPS URI."
++      );
+     }
+-    else if (openId4VcTokenIssuer.method === 'x5c') {
+-        const issuer = openId4VcTokenIssuer.issuer;
+-        const leafCertificate = core_1.X509Service.getLeafCertificate(agentContext, {
+-            certificateChain: openId4VcTokenIssuer.x5c,
+-        });
+-        if (!issuer.startsWith('https://')) {
+-            throw new core_1.CredoError('The X509 certificate issuer must be a HTTPS URI.');
+-        }
+-        if ((_b = leafCertificate.sanUriNames) === null || _b === void 0 ? void 0 : _b.includes(issuer)) {
+-            return Object.assign(Object.assign({}, openId4VcTokenIssuer), { clientIdScheme: 'x509_san_uri' });
+-        }
+-        else {
+-            if (!((_c = leafCertificate.sanDnsNames) === null || _c === void 0 ? void 0 : _c.includes((0, core_1.getDomainFromUrl)(issuer)))) {
+-                throw new Error(`The 'iss' claim in the payload does not match a 'SAN-URI' or 'SAN-DNS' name in the x5c certificate.`);
+-            }
+-            return Object.assign(Object.assign({}, openId4VcTokenIssuer), { clientIdScheme: 'x509_san_dns' });
+-        }
++
++    const jwk = (0, core_1.getJwkFromKey)(leafCertificate.publicKey);
++    const alg = jwk.supportedSignatureAlgorithms[0]
++    if (
++      (_b = leafCertificate.sanUriNames) === null || _b === void 0
++        ? void 0
++        : _b.includes(issuer)
++    ) {
++      return Object.assign(Object.assign({}, openId4VcTokenIssuer), {
++        clientIdScheme: "x509_san_uri",
++        alg: alg,
++      });
++    } else {
++      if (
++        !((_c = leafCertificate.sanDnsNames) === null || _c === void 0
++          ? void 0
++          : _c.includes((0, core_1.getDomainFromUrl)(issuer)))
++      ) {
++        throw new Error(
++          `The 'iss' claim in the payload does not match a 'SAN-URI' or 'SAN-DNS' name in the x5c certificate.`
++        );
++      }
++      return Object.assign(Object.assign({}, openId4VcTokenIssuer), {
++        clientIdScheme: "x509_san_dns",
++        alg: alg,
++      });
+     }
+-    return openId4VcTokenIssuer;
++  }
++  return openId4VcTokenIssuer;
+ }
+ function getProofTypeFromKey(agentContext, key) {
+-    const signatureSuiteRegistry = agentContext.dependencyManager.resolve(core_1.SignatureSuiteRegistry);
+-    const supportedSignatureSuites = signatureSuiteRegistry.getAllByKeyType(key.keyType);
+-    if (supportedSignatureSuites.length === 0) {
+-        throw new core_1.CredoError(`Couldn't find a supported signature suite for the given key type '${key.keyType}'.`);
+-    }
+-    return supportedSignatureSuites[0].proofType;
++  const signatureSuiteRegistry = agentContext.dependencyManager.resolve(
++    core_1.SignatureSuiteRegistry
++  );
++  const supportedSignatureSuites = signatureSuiteRegistry.getAllByKeyType(
++    key.keyType
++  );
++  if (supportedSignatureSuites.length === 0) {
++    throw new core_1.CredoError(
```

Comment:
Due to the formatting it's very hard to see what changed, and it will also  break much faster as any change to the file will now result in merge conflicts

Reply:
It is already in PR so this patch is very temporary. I just wanted to get going and not wait for the full dPoP PR.